src/compiler/parser.ts

```diff
@@ -1881,6 +1881,11 @@ namespace ts {
                     if (considerSemicolonAsDelimiter && token() === SyntaxKind.SemicolonToken && !scanner.hasPrecedingLineBreak()) {
                         nextToken();
                     }
+                    else if (isJSDocParameterStart() && parsingContext & (1 << ParsingContext.Parameters)) {

```

Comment:
`parseDelimitedList` is called in a lot of different places, so it seems wrong to put something in here that would only be used in one particular situation. It also looks like this would be the only place in the parser that we compare `parsingContext` to a particular value. Could this be moved into the `parseElement` handler?